sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -92,16 +92,30 @@ case class InsertIntoHadoopFsRelationCommand(
 
     var initialMatchingPartitions: Seq[TablePartitionSpec] = Nil
     var customPartitionLocations: Map[TablePartitionSpec, String] = Map.empty
-    var matchingPartitions: Seq[CatalogTablePartition] = Seq.empty
 
     // When partitions are tracked by the catalog, compute all custom partition locations that
     // may be relevant to the insertion job.
     if (partitionsTrackedByCatalog) {
```

Comment:
Based the code on [HiveClientImpl.getTableOption](https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L546) and [CatalogTable](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala#L250), it seems that we won't run into this code while using hive metastore.

Reply:
Thanks for your response. `tracksPartitionsInCatalog` is set true in [HiveExternalCatalog.restoreHiveSerdeTable](https://github.com/apache/spark/blob/52e7602344036a649bb19ed642dbd74bc7ee9cb1/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala#L800)